app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt

```diff
@@ -441,14 +450,21 @@ private class ReleaseMetricController(
     override fun start() {
         if (!isTelemetryEnabled.invoke() || initialized) { return }
 
-        services.forEach { it.start() }
+        managedScope = CoroutineScope(Dispatchers.IO)
+        services.forEach { it.start(managedScope!!) }
         initialized = true
     }
 
     override fun stop() {
         if (!initialized) { return }
 
         services.forEach { it.stop() }
+        run {
```

Comment:
nit: can this just be `managedScope?.cancel()`? fwiw, if I understand the intent, I haven't seen `run` used to group code before: usually whitespace is sufficient. Then again, I almost never see `run` used :P